addon/routes/oidc-authentication.js

```diff
@@ -122,9 +122,18 @@ export default class OIDCAuthenticationRoute extends Route {
      * the login he can be sent to the initial destination.
      */
     if (!this.session.data.nextURL) {
-      const route = this.session.attemptedTransition?.to.name;
-      const url = route && this.router.urlFor(route);
-      this.session.set("data.nextURL", url);
+      const transition = this.session.attemptedTransition?.to;
+
+      if (transition) {
+        const queryParams = Object.entries(transition.queryParams)
+          .map(([key, value]) => `${key}=${encodeURIComponent(value)}`)
+          .join("&");
+        const url = [this.router.urlFor(transition.name), queryParams]
+          .filter(Boolean)
+          .join("?");
+
+        this.session.set("data.nextURL", url);
```

Comment:
Have you tried using `this.session.attemptedTransition.intent.url`?

Reply:
That's exactly what I was looking for, it works. Is this documented somewhere in ember? Because I couldn't find anything.

Reply:
Sadly, it's not documented anywhere but on stackoverflow :grin:

Reply:
Ahh I see :D Will it be safe to use? It might break with an ember update, no?

Reply:
It's been there forever and it's used in a lot of addons (including `ember-simple-auth`). If it breaks we'll fix it then.

Reply:
ok, done :+1: